core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -624,7 +624,19 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
       } yield ()
     }
 
-  def assertNoChildrenBlockingWorkspaceDeletion(workspace: Workspace): Future[Unit] = for {
+  def assertNoGoogleChildrenBlockingWorkspaceDeletion(workspace: Workspace): Future[Unit] = for {
```

Comment:
I think there's still a problem when the workspace is a v1 workspace in which case the workspace won't have a google project resource under it in sam but that google project might have resouces under it (eg leo stuff)

Reply:
Presumably we rapidly no longer care about v1 workspaces though.

Reply:
I'm not sure that we can stop caring about them until they're officially unsupported at the end of the month unfortunately. One way to address this would be to change the line where we list out the `googleProjectChildren` to only do so if there is a `google-project` that is a child of the workspace. This would be more correct anyways since any children of the `google-project` resource would only block deletion if the `google-project` was a child of the workspace. So it'd be something like:

```
    workspaceChildren <- samDAO
      .listResourceChildren(SamResourceTypeNames.workspace, workspace.workspaceId, ctx)
    maybeWorkspaceGoogleProject = workspaceChildren.find(_.resourceTypeName == SamResourceTypeNames.googleProject.value)
    googleProjectChildren <- maybeWorkspaceGoogleProject.traverse { workspaceGoogleProject => samDAO.listResourceChildren(SamResourceTypeNames.googleProject, workspaceGoogleProject.resourceId, ctx)}
    blockingChildren = workspaceChildren.filter(c =>
      c.resourceTypeName != SamResourceTypeNames.googleProject.value || workspace.googleProjectId.value != c.resourceId
    ).toList ::: googleProjectChildren.getOrElse(Seq.empty).toList
```

Reply:
exactly